docs/quickstart.md

```diff
@@ -0,0 +1,760 @@
+                Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
+    ```
+In this example we added two methods `hello_name` and `hello` next to each other. We added `Router` class that allow us to register what method should be called when specific request comes. 
+We created instance of it and added configuration with mapping between processing methods and http request method. 
+In lambda handler we call router instance get method to fetch reference to processing method (`hello` or `hello_name`) that will handle the request. 
+Finally we run this method and return result to API Gateway.
+
+Thanks to this approach, we can simplify our infrastructure configuration since we added all API Gateway path under `HelloWorldFunction` event section. 
+Downside of current approach is that we need to add custom class and with every new method or path we want to serve we need to add explicit configuration. 
+Also, we need to understand API Gateway request event structure in order to decapsulate requested path and http method. 
+In order to mitigate this downsides we use Lambda Powertools.
+
+=== "app.py"
+
+    ```python hl_lines="1 4 7 12 18"
+    from aws_lambda_powertools.event_handler.api_gateway import ApiGatewayResolver
+
+
+    app = ApiGatewayResolver()
+
+
+    @app.get("/hello/<name>")
+    def hello_name(name):
+        return {"statusCode": 200, "body": {"message": f"hello {name}!"}}
+
+
+    @app.get("/hello")
+    def hello():
+        return {"statusCode": 200, "body": {"message": "hello unknown!"}}
+
+
+    def lambda_handler(event, context):
+        return app.resolve(event, context)
+    ```
+=== "requirements.txt"
+
+    ```bash
+    aws-lambda-powertools
+    ```
+
+Since we used powertools api_gateway event handler we don't need to know internal event structure. All we do we instantiate `ApiGatewayResolver` class and we call it in our `lambda_handler`.
+Since we added route annotation as a decorator for our methods we can directly use parameters passed in request.
+In order for SAM to build our package correctly we specified lambda powertools package in our requirements file. 
+It is then automatically downloaded by SAM and added to lambda zip file. 
+
+!!! tip
+    If you are interested in learning how python decorators work under the hood you can follow great article from [Real Python](https://realpython.com/primer-on-python-decorators/).
+## Structured Logging
+In next step you have been tasked to propose production grade logging capabilities to your lambda code.
+You decided to follow [structured logging approach](https://docs.aws.amazon.com/lambda/latest/operatorguide/parse-logs.html) which means that you don’t write hard-to-parse and hard-to-keep-consistent prose in your logs but that you log events that happen in a context instead. 
+We want our log event to be in JSON format. This will allow us to search through it easily in CloudWatch Logs or Cloudwatch Insight.
```

Comment:
This is the key information presented in the 3rd sentence, let's pull it higher and then explain why it is important: hard-to-prase, link for more resources etc.

Reply:
Refactored to:
`In the next step, you have been given the task of proposing production quality logging capabilities to your lambda code.
We want our log event to be in a JSON format. Also, You decided to follow [structured logging approach](https://docs.aws.amazon.com/lambda/latest/operatorguide/parse-logs.html). In a result, we expect easy to search, consistent logs containing enough context and data to analyse the status of our system. We can take advantage of CloudWatch Logs and Cloudwatch Insight for this purpose.`
Let me know what do you think